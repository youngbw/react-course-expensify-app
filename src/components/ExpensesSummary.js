import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h2>
            Viewing {props.expenses.length}
            {
                props.expenses.length === 1 ? ' expense ' : ' expenses '
            }  
            totalling {numeral(props.total / 100).format('$0,0.00')}
        </h2>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        total: expensesTotal(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);