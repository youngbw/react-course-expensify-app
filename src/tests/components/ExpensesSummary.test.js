import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render 1 expense total correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} total={10000} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render multiple expenses total correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} total={238429834} />);
    expect(wrapper).toMatchSnapshot();
});