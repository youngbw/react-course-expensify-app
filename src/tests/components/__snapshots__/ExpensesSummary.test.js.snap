// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense total correctly 1`] = `
<div>
  <h2>
    Viewing 
    1
     expense 
    totalling 
    $100.00
  </h2>
</div>
`;

exports[`should render multiple expenses total correctly 1`] = `
<div>
  <h2>
    Viewing 
    3
     expenses 
    totalling 
    $2,384,298.34
  </h2>
</div>
`;
